### PR TITLE
LibGfx+icc: Read cicpType

### PR DESCRIPTION
**Pull request by nico**
This is a very new tag used for HDR content. The only files I know that use it are the jpegs on https://ccameron-chromium.github.io/hdr-jpeg/ But they have an invalid ICC creation date, so `icc` can't process them. (Commenting out the check for that does allow to print them.)

If the CIPC tag is present, it takes precedence about the actual data in the profile and from what I understand, the ICC profile is basically ignored. See https://www.color.org/events/HDR_experts.xalter for background, in particular
https://www.color.org/hdr/02-Luke_Wallis.pdf (but the other talks are very interesting too).

(PNG also has a cICP chunk that's supposed to take precedence over iCCP.)

### CODE REVIEW

**Comment by nico** [comment]
```
% Build/lagom/icc ~/Downloads/pescariello0-hlg-rec2020.jpeg
[...]
cicpTag ('cicp'): type 'cicp', offset 564, size 12
    color primaries: 9
    transfer characteristics: 18
    matrix coefficients: 0
    video full range flag: 1
```

**Comment by Zaggy1024** [comment]
What are your thoughts on pretty printing these in `icc` using the helpers in CodingIndependentCodePoints.h? I'm sure that it should print the raw value first, so maybe something like:

```
% Build/lagom/icc ~/Downloads/pescariello0-hlg-rec2020.jpeg
[...]
cicpTag ('cicp'): type 'cicp', offset 564, size 12
    color primaries: 9 (BT.2020)
    transfer characteristics: 18 (ARIB STD-B67 (HLG, BT.2100))
    matrix coefficients: 0 (Identity)
    video full range flag: 1
```

Although now I'm noticing that the parentheses that those `...to_string()` functions create don't fit into this too well, perhaps I should go back and rethink the text for those.

**Comment by nico** [comment]
Great idea, done!

What do you think about adding a `color_range_to_string()` function as well? (not in this PR)

What do you think about making these functions return an Optional<StringView>? Then the code here could only print the code for the "Reserved" case, but not for the known values. (Also not in this PR.)

Medium-term, we might want to move the LibVideo color code to LibGfx, so that if we use CICPs for color-managing images, we don't end up with a dependency cycle between LibGfx and LibVideo.

**Comment by nico** [comment]
New screenshot:

```
    color primaries: 9 - BT.2020
    transfer characteristics: 18 - ARIB STD-B67 (HLG, BT.2100)
    matrix coefficients: 0 - Identity
    video full range flag: 1
```

File: Userland/Utilities/icc.cpp [hunk 1]
```diff
@@ -125,7 +126,16 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         }
         tag_data_to_first_signature.set(tag_data, tag_signature);
 
-        if (tag_data->type() == Gfx::ICC::CurveTagData::Type) {
+        if (tag_data->type() == Gfx::ICC::CicpTagData::Type) {
+            auto& cicp = static_cast<Gfx::ICC::CicpTagData&>(*tag_data);
+            outln("    color primaries: {} - {}", cicp.color_primaries(),
+                Video::color_primaries_to_string((Video::ColorPrimaries)cicp.color_primaries()));
```
Comment (linusg) [inline Userland/Utilities/icc.cpp:132]: This feels like something that should live in LibGfx instead if ICC profiles use it as well, and then LibVideo would use that (for later).

Comment (nico) [reply Userland/Utilities/icc.cpp:132]: https://github.com/SerenityOS/serenity/pull/17369#issuecomment-1422640125 – agreed!

Comment (Zaggy1024) [reply Userland/Utilities/icc.cpp:132]: I concur!

**Pull request merged**